ckanext/validation/logic.py

```diff
@@ -174,6 +175,43 @@ def resource_validation_show(context, data_dict):
 
     return _validation_dictize(validation)
 
+def resource_table_schema_infer(context, data_dict):
+    '''
+    Use frictionless framework to infer a resource schema
+    '''
+
+    t.check_access('resource_create', context, data_dict)
+
+    if not data_dict.get(u'resource_id'):
+        raise t.ValidationError({u'resource_id': u'Missing value'})
+
+    store_schema = data_dict.get('store_schema', True)
+
+    resource = t.get_action('resource_show')(
+        {}, {u'id': data_dict['resource_id']})
+
+    source = None
```

Comment:
If the resource['format'] is not one of `ckanext.validation.formats` we should raise a `ValidationError` with : "Not a valid format to infer the resource schema"